### DISCUSSION TITLE
How to change logs level and format?

### DISCUSSION THREAD

**Discussion by Hronom** [crossplane/crossplane#4163] [Q&A] [answered]
How to set logs level to `warn` for example?

And how to set logs format to `json`?

**Comment by negz** [comment, marked as answer]
We don't have any warning logs in Crossplane - they're only info and debug. We currently log almost nothing at info level.

https://github.com/crossplane/crossplane/blob/master/contributing/guide-observability.md describes our logging philosophy. In short, we mostly treat logs as a tool to debug Crossplane when something is going really wrong. For day-to-day observability we prefer to emit Kubernetes events and update resource status conditions. The rationale is that if your MR (for example) isn't working we should not assume you have access to the Crossplane pod logs to figure out what is going wrong.

The default log format (when running at the default log level - info) is JSON. When you change Crossplane to debug logs it outputs in a hybrid format. (i.e. Plaintext messages with a bunch of JSON metadata attached.)

**Reply by triThirty** [reply]
Hi @negz , is there any way to make Json format at debug level?
